boards/PDM/Src/Io/Io_CurrentCheck.c

```diff
@@ -0,0 +1,139 @@
+    }
+}
+
+float Io_CurrentInRangeCheck_GetAux2Current(void)
+{
+    // TODO: Fix hard-coded value once the ADC is configured
+    return 0.5f;
+}
+
+void Io_CurrentInRangeCheck_Aux2ErrorCallback(enum InRangeCheck_Status status)
+{
+    if (status == VALUE_UNDERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_UNDERCURRENT_AUX_2(_can_tx, true);
+    }
+    else if (status == VALUE_OVERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_OVERCURRENT_AUX_2(_can_tx, true);
+    }
+}
+
+float Io_CurrentInRangeCheck_GetLeftInverterCurrent(void)
+{
+    // TODO: Fix hard-coded value once the ADC is configured
+    return 0.5f;
+}
+
+void Io_CurrentInRangeCheck_LeftInverterErrorCallback(
+    enum InRangeCheck_Status status)
+{
+    if (status == VALUE_UNDERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_UNDERCURRENT_LEFT_INVERTER(_can_tx, true);
+    }
+    else if (status == VALUE_OVERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_OVERCURRENT_LEFT_INVERTER(_can_tx, true);
+    }
+}
+
+float Io_CurrentInRangeCheck_GetRightInverterCurrent(void)
+{
+    // TODO: Fix hard-coded value once the ADC is configured
+    return 0.5f;
+}
+
+void Io_CurrentInRangeCheck_RightInverterErrorCallback(
+    enum InRangeCheck_Status status)
+{
+    if (status == VALUE_UNDERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_UNDERCURRENT_RIGHT_INVERTER(_can_tx, true);
+    }
+    else if (status == VALUE_OVERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_OVERCURRENT_RIGHT_INVERTER(_can_tx, true);
+    }
+}
+
+float Io_CurrentInRangeCheck_GetEnergyMeterCurrent(void)
+{
+    // TODO: Fix hard-coded value once the ADC is configured
+    return 0.5f;
+}
+
+void Io_CurrentInRangeCheck_EnergyMeterErrorCallback(
+    enum InRangeCheck_Status status)
+{
+    if (status == VALUE_UNDERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_UNDERCURRENT_ENERGY_METER(_can_tx, true);
+    }
+    else if (status == VALUE_OVERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_OVERCURRENT_ENERGY_METER(_can_tx, true);
+    }
+}
+
+float Io_CurrentInRangeCheck_GetCanCurrent(void)
+{
+    // TODO: Fix hard-coded value once the ADC is configured
+    return 0.5f;
+}
+
+void Io_CurrentInRangeCheck_CanErrorCallback(enum InRangeCheck_Status status)
+{
+    if (status == VALUE_UNDERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_UNDERCURRENT_CAN(_can_tx, true);
+    }
+    else if (status == VALUE_OVERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_OVERCURRENT_CAN(_can_tx, true);
+    }
+}
+
+float Io_CurrentInRangeCheck_GetAirShutdownCurrent(void)
+{
+    // TODO: Fix hard-coded value once the ADC is configured
+    return 0.5f;
+}
+
+void Io_CurrentInRangeCheck_AirShutdownErrorCallback(
+    enum InRangeCheck_Status status)
+{
+    if (status == VALUE_UNDERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_UNDERCURRENT_AIR_SHUTDOWN(_can_tx, true);
+    }
+    else if (status == VALUE_OVERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_OVERCURRENT_AIR_SHUTDOWN(_can_tx, true);
+    }
```

Comment:
bit of a nit, but generally I prefer `switch` statements instead of `if` statements for enums because you can do a compile-time check that all cases of a given enum are being properly handled (as long as you don't have a `default` case).

Reply:
This really is a nit though, I don't feel that strongly about this, up to you if you want to change it.

Reply:
Turned into switch statements !